testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/core/spi/ResourceClassProcessorNotAppliedTest.java

```diff
@@ -0,0 +1,87 @@
+package org.jboss.resteasy.test.core.spi;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.test.core.spi.resource.ResourceClassProcessorNotAppliedImplementation;
+import org.jboss.resteasy.test.core.spi.resource.ResourceClassProcessorPureEndPoint;
+import org.jboss.resteasy.util.HttpResponseCodes;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
+/**
+ * @tpSubChapter ResourceClassProcessor SPI
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails ResourceClassProcessor should not be used in some case
+ * @tpSince RESTEasy 3.6
+ */
+@RunWith(Arquillian.class)
+public class ResourceClassProcessorNotAppliedTest {
+
+    protected static final Logger logger = Logger.getLogger(ResourceClassProcessorNotAppliedTest.class.getName());
+
+    private static List<String> visitedProcessors = new ArrayList<>();
+
+    public static void addToVisitedProcessors(String item) {
```

Comment:
Shouldn't this be synchronized?

Reply:
AFAICS no, because processors are applied in sequence, see [java code](https://github.com/resteasy/Resteasy/pull/1449/files#diff-a7cd5f8090c07c60ab3f97314b8091bbR1064). But we may want to be super safe and synchronize this. So I add `synchronized` keyword ...